bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/discovery/SunspecDiscoveryParticipant.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.sunspec.internal.discovery;
+
+import static org.openhab.binding.modbus.sunspec.internal.SunSpecConstants.SUPPORTED_THING_TYPES_UIDS;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.openhab.binding.modbus.discovery.ModbusDiscoveryListener;
+import org.openhab.binding.modbus.discovery.ModbusDiscoveryParticipant;
+import org.openhab.binding.modbus.handler.EndpointNotInitializedException;
+import org.openhab.binding.modbus.handler.ModbusEndpointThingHandler;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for sunspec
+ *
+ * @author Nagy Attila Gabor - initial contribution
+ *
+ */
+@Component(immediate = true)
+@NonNullByDefault
+public class SunspecDiscoveryParticipant implements ModbusDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(SunspecDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return new HashSet<ThingTypeUID>(SUPPORTED_THING_TYPES_UIDS.values());
```

Comment:
Can you make this set static?

Reply:
I'm afraid I don't understand you request. The SUPPORTED_THING_TYPES_UIDS is a static map in the SunSpecConstants class. All I do here is that I return the values as a set as requested by the interface.
Creating a second static set to support this method would be a duplication of the configuration.
Can you please give me a hint what you would like better?

Reply:
Actually, I think you are correct here. I'm not entirely sure what I saw here when I recommended the change. I need to stop doing code reviews when I'm tired.